doc/specs/#980 - Apps and Features entries version mapping.md

```diff
@@ -0,0 +1,193 @@
+```text
+Winget version:       1.0.0                   2.0.0                  3.0.0
+ARP version range:    [10.0, 10.5]            [7.0, 7.5]             [13.0, 13.7]
+```
+
+ARP version `10.4` will be mapped as Winget version `1.0.0`.  
+ARP version `13.4` will be mapped as Winget version `3.0.0`.  
+ARP version `9.4` will be mapped as Winget version `Unknown`.
+
+#### Full version mapping performed
+For packages with ARP version ordered in the same order of Winget version, full mapping will be performed. For example:
+```text
+Winget version:       1.0.0                   2.0.0                  3.0.0             4.0.0
+ARP version range:    [10.0, 10.5]            [Not provided]         [12.0, 12.5]      [13.0, 13.5]
+```
+
+Winget will perform following mapping:
+1. Check if ARP version fall within one of the ARP version range
+2. Try to find the closest ARP version range it's less than, this is higher priority than greater than mapping because this mapping will mostly be used for package upgrade applicability check.
+3. Try to find the closes ARP version range it's greater than
+
+A special "less than"("< ") and "greater than"("> ") version concept will be used. This is to indicate the version is less than or greater than the closest version Winget is known of. UI/UX change to the output is described in later UI/UX section.
+
+For above version mapping:  
+ARP version `10.0` will be mapped as Winget version `1.0.0`.  
+ARP version `11.7` will be mapped as less than Winget version 3.0.0(`< 3.0.0`).  
+ARP version `12.7` will be mapped as less than Winget version 4.0.0(`< 4.0.0`).  
+ARP version `14.0` will be mapped as greater than Winget version 4.0.0(`> 4.0.0`).  
+ARP version `2.0.0` will be mapped as less than Winget version 1.0.0(`< 1.0.0`)(Once ARP version mapping logic is applied, Winget will only look at ARP version range, though `2.0.0` exactly matches one Winget version).
+
+For version comparison with special "less than" or "greater than", the "less than" or "greater than" only applies when compared to the specific version, for comparing to other versions, "less than" or "greater than" could be considered as ignored.
+
+For example:  
+Version `< 3.0` is less than version `3.0`  
+Version `< 3.0` is greater than version `2.9`(because `3.0` is greater than `2.9`)  
+Version `< 3.0` is less than version `4.0`  
+Version `< 3.0` is less than version `> 3.0`  
+Version `> 3.0` is greater than version `3.0`  
+Version `> 3.0` is less than version `3.1`  (because `3.0` is less than `3.1`)  
+Version `> 3.0` is greater than version `2.9`  
+
+**Note:** It is recommended for package authors to update ARP version info for all package versions of a package for better version mapping if this feature is to be used for a package.
```

Comment:
What if a publisher updates the ARP version but not the Marketing version? How should that be handled? I'm specifically thinking more from a manifest-creation and winget-pkgs standpoint here

Reply:
There can only be one package (Marketing) version manifest, so it would be an update to that, replacing it.  This would be the case for a bug fix only update that is intended to fully replace the existing version.